proxy.go

```diff
@@ -29,7 +29,7 @@ func (fn netDialerFunc) DialContext(ctx context.Context, network, addr string) (
 }
 
 func proxyFromURL(proxyURL *url.URL, forwardDial netDialerFunc) (netDialerFunc, error) {
-	if proxyURL.Scheme == "http" {
+	if proxyURL.Scheme == "http" || proxyURL.Scheme == "https" {
```

Comment:
is it there other possibility? doesn't this mean, we always use the `httpProxyDialer` ?

Reply:
Yes--I'll always use the `httpProxyDialer`. The difference is the `forwardDial`, which now will also perform the TLS handshake if necessary.

Reply:
my point is that proxyFromURL now is just `return (&httpProxyDialer{proxyURL: proxyURL, forwardDial: forwardDial}).DialContext, nil` the other code will never be executed so we should not keep it if that is the case, or can lead to confusion

Reply:
oh, good point, I was not considering those